### ISSUE TITLE
Not able to see types

### ISSUE THREAD

**Issue by agarwalayush** [begriffs/haskell-vim-now#74]
I am not able to see the types of expression under cursor or lines (using `,ht`). 
Error on using `,ht`  is   `ghcmod#command#type: Cannot guess type`
and similarly on using `,hT`  is `ghcmod#command#type_insert: Cannot guess type` 
It is not that above is not working on some particular instance, it has not worked ever since I installed `vim-haskell-now`.
I tried running `ghc-mod check /path_to_file`, but there was no error.

Also `,hc` doesn't work. Maybe I guess some problem with `syntastic`.

**Comment by begriffs** [comment]
Are you using stack to build the project in question or cabal (with a sandbox)?

**Comment by MaikKlein** [comment]
I have the same problem, I am using stack to build the project.

```
stack new foo
cd foo
stack setup
stack build
```

**Comment by begriffs** [comment]
How recently did you run the haskell vim now installer? It's been undergoing some changes lately. If it's been a while you should try running it again to update your system.

**Comment by MaikKlein** [comment]
I ran it 1 hour ago, I am on Ubuntu 14.04.

**Comment by begriffs** [comment]
I suspect it's caused by the project specifying ghc-mod 5.3 rather than 5.4. I'll try to reproduce on a clean ubuntu machine.

**Comment by MaikKlein** [comment]
The current version that I have is 5.3

```
> $ ~/.local/bin/ghc-mod --version                                                                                                                
ghc-mod version 5.3.0.0 compiled by GHC 7.10.2

> $ ~/.local/bin/ghc-mod find someFunc                                                                                                            
cabal: At least the following dependencies are missing:
base >=4.7 && <5
ghc-mod: readCreateProcess: cabal "configure" "--with-ghc=ghc" (exit 1): failed
ghc-mod: readCreateProcess: /home/maik/.local/bin/ghc-mod "dumpsym" "/tmp/ghc-mod-home-maik-projects-haskell-test10534" (exit 1): failed
```

I tried to update it with

```
> $ stack install ghc-mod                                                                                                                         
Run from outside a project, using implicit global project config
Using resolver: lts-3.10 from implicit global project's config file: /home/maik/.stack/global-project/stack.yaml
While constructing the BuildPlan the following exceptions were encountered:

--  Failure when adding dependencies:    
      cabal-helper: needed (==0.6.* && >=0.6.0.0), 0.5.3.0 found (latest is 0.6.1.0)
    needed for package: ghc-mod-5.4.0.0

> $ stack install cabal-helper                                                                                                                    
Run from outside a project, using implicit global project config
Using resolver: lts-3.10 from implicit global project's config file: /home/maik/.stack/global-project/stack.yaml
Couldn't find executable cabal-helper-wrapper in directory /home/maik/.stack/global-project/.stack-work/install/x86_64-linux/lts-3.10/7.10.2/bin/
```

**Comment by begriffs** [comment]
Yeah in order to hold ghc-mod back at version 5.3 I had to set the global stack dependencies back at cabal-helper-0.5.3.0. I'm experimenting with changing both restrictions. I held the versions back because of https://github.com/scrooloose/syntastic/issues/1576 but I think if I had to choose that I'd rather be able to query for types than have syntastic error markers. The versions of everything are in flux so it's a balancing act to have everything in flux.

We'll get the syntastic thing fixed eventually. I may have to try helping to maintain the ghc-mod syntastic checker myself...
